YAML: 1.0

preprocess:
  point_filter_num: 1
  lidar_type: 2  # 1-AVIA 2-velodyne 3-ouster  4-robosense 5-pandar 6-livox
  blind: 0.1


common:
  imu_topic: /livox/imu #/imu_raw   /gps/gtimu
  lid_topic: /velodyne_points  #/points_raw  /rslidar_points /velodyne_points

mapping:
  extrinsic_est_en: true
  extrinsic_T: [ 0.0, 0.0, 0.0]
  extrinsic_R: [ 1, 0, 0, 0, 1, 0, 0, 0, 1]

delay_time: 0.3

odometry:
  surf_res: 0.4
  log_print: false
  max_num_iteration: 8
  # ct_icp
  icpmodel: CT_POINT_TO_PLANE                    # Options: [CT_POINT_TO_PLANE, POINT_TO_PLANE]
  size_voxel_map: 0.4                         # The voxel size of in the voxel map
  min_distance_points: 0.08
  max_num_points_in_voxel: 20                 # The maximum number of points per voxel of the map
  max_distance: 500.0                        # The threshold of the distance to suppress voxels from the map
  weight_alpha: 0.9
  weight_neighborhood: 0.1
  max_dist_to_plane_icp: 0.3
  init_num_frames: 20
  voxel_neighborhood: 1
  max_number_neighbors: 20
  threshold_voxel_occupancy: 1
  estimate_normal_from_neighborhood: true
  min_number_neighbors: 20                    # The minimum number of neighbor points to define a valid neighborhood
  power_planarity: 2.0
  num_closest_neighbors: 1

  sampling_rate: 1.5
  max_num_residuals: 1500
  min_num_residuals: 100
  motion_compensation: CONTINUOUS #NONE, CONSTANT_VELOCITY, ITERATIVE, CONTINUOUS
  beta_location_consistency: 1.0
  beta_orientation_consistency: 1.0
  beta_constant_velocity: 0.0
  beta_small_velocity: 0.0

  thres_translation_norm: 0.01            # m
  thres_orientation_norm: 0.1             # deg